float32 baseAng
float32 mainAng
float32 secAng
float32 toolAng